src/lib/storage/run_length_segment/run_length_segment_iterable.hpp

```diff
@@ -136,8 +136,8 @@ class RunLengthSegmentIterable : public PointAccessibleSegmentIterable<RunLength
 
     explicit PointAccessIterator(const pmr_vector<T>* values, const pmr_vector<bool>* null_values,
                                  const pmr_vector<ChunkOffset>* end_positions,
-                                 const PosList::const_iterator position_filter_begin,
-                                 PosList::const_iterator position_filter_it)
+                                 const PosList::const_iterator&& position_filter_begin,
+                                 PosList::const_iterator&& position_filter_it)
         : BasePointAccessSegmentIterator<PointAccessIterator, SegmentPosition<T>>{std::move(position_filter_begin),
```

Comment:
This was already broken before. I am not sure why `std::move` on a const-qualified type is legal.